### DISCUSSION TITLE
Naming render groups and items in the graphics compositor

### DISCUSSION THREAD

**Discussion by ghost** [stride3d/stride#1467] [General]
So far I have found two places where I believe given custom friendly names would make it easier to manage.

First one is render groups.
![image](https://user-images.githubusercontent.com/104224536/180432135-c2087479-826d-45e2-af1d-eda954281242.png)

Second one is the Graphics Compositor where all items such as the "Forward renderer" appear with the same name even though they are separate instances with different configurations, actually two of them are the same instance but you can't tell from the screenshot.
![image](https://user-images.githubusercontent.com/104224536/180432273-5c12f5ff-218a-4d1d-828c-ee46207dc723.png)
A use case would be to experiment with different settings. You can have two "Forward renderer" that you swap back and forth, then it's very useful to know which one is the main one and which one is set up for testing other settings.

Perhaps it's already possible. If not, would it be desirable?

**Comment by manio143** [comment]
So at the moment things like Render group and physics ColliderGroup are enums with a constant set of values (int32 bit mask). It would be cool if the user could define aliases for some of those enum values to be shown instead of their field names. That's an actual new feature that'd have to be done as we need to consider what the UX for this makes the most sense.

For the forward renderer - since it's a list of instances we should be able to see if it's ToString() that's being shown and modify to allow the user to set a name on a renderer and show that instead.

**Reply by ghost** [reply]
Would UX be a concern for implementing the first version of this?

**Reply by tebjan** [reply]
You can go ahead and propose a UX, but we can't help much with the implementation because this issue has lower priority for the engine as other things.
